### PR TITLE
ast: Unmerge structures for associated items and foreign items

### PR DESCRIPTION
**Pull request by petrochenkov**
Follow-up to https://github.com/rust-lang/rust/pull/69194.
r? @Centril

### CODE REVIEW

File: src/librustc_ast_pretty/pprust.rs [hunk 1]
```diff
@@ -1413,8 +1398,29 @@ impl<'a> State<'a> {
     }
 
     crate fn print_assoc_item(&mut self, item: &ast::AssocItem) {
```
Comment (Centril) [inline src/librustc_ast_pretty/pprust.rs]: Hmm; the duplication here and in `visit.rs` isn't great. Let's follow up with the common structs to get some more code reuse.

**Comment by JohnTitor** [comment]
Seems spurious.
@bors retry

**Pull request merged**